src/streaming_decoder.rs

```diff
@@ -39,10 +39,35 @@ impl<READ: Read> StreamingDecoder<READ, FrameDecoder> {
         decoder.init(&mut source)?;
         Ok(StreamingDecoder { decoder, source })
     }
+}
 
-    pub fn inner(self) -> FrameDecoder {
+impl<READ: Read, DEC: BorrowMut<FrameDecoder>> StreamingDecoder<READ, DEC> {
+    /// Destructures this object into the inner [FrameDecoder].
+    pub fn inner(self) -> DEC {
```

Comment:
I think this should be called `into_inner` for consistency with `into_inner_parts`

Reply:
Maybe into_frame_decoder?

Reply:
I like that :+1: